test/e2e/performanceprofile/functests/utils/tuned/tuned.go

```diff
@@ -84,38 +84,36 @@ func GetPod(ctx context.Context, node *corev1.Node) (*corev1.Pod, error) {
 }
 
 func WaitForStalldTo(run bool, interval, timeout time.Duration, node *corev1.Node) error {
-	return wait.Poll(interval, timeout, func() (bool, error) {
-		cmd := []string{"/bin/bash", "-c", "pidof stalld || true"}
+	return wait.PollImmediate(interval, timeout, func() (bool, error) {
+		cmd := []string{"/bin/bash", "-c", "pidof stalld"}
 		stalldPid, err := nodes.ExecCommandOnNode(cmd, node)
 		if err != nil {
-			return false, fmt.Errorf("failed to execute command %q on node: %q; %w", cmd, node.Name, err)
+			klog.Warningf("pidof stalld command on node=%q failed to execute with the following error: %v", node.Name, err)
 		}
-
 		_, err = strconv.Atoi(stalldPid)
 		if !run { // we don't want stalld to run
 			if err == nil {
-				return false, fmt.Errorf("node=%q stalld_pid=%q stalld is running when it shouldn't", node.Name, stalldPid)
+				return false, nil
 			}
+			return true, nil
 		}
 		// we want stalld to run
 		if err != nil {
-			klog.Warningf("node=%q stalld_pid=%q is not a valid pid number: %v", node.Name, stalldPid, err)
 			return false, nil
 		}
 		return true, nil
 	})
 }
 
 func CheckParameters(node *corev1.Node, sysctlMap map[string]string, kernelParameters []string, stalld, rtkernel bool) {
-	cmd := []string{"/bin/bash", "-c", "pidof stalld || true"}
+	cmd := []string{"/bin/bash", "-c", "pidof stalld"}
 	By(fmt.Sprintf("Executing %q", cmd))
-	stalldPid, err := nodes.ExecCommandOnNode(cmd, node)
-	ExpectWithOffset(1, err).ToNot(HaveOccurred(), "failed to execute command %q on node: %q; %w", cmd, node.Name, err)
+	stalldPid, _ := nodes.ExecCommandOnNode(cmd, node)
```

Comment:
This is still an issue, we can't ignore the error here

Reply:
ok, so I might bring back the pipe format that was non consistently failing  maybe due to the wg issue we had and test it

Reply:
This is not an actual piping BTW, just an OR operator.
@jmencak Could you please elaborate when did you encounter issues with this `||` operator when it's part of a command executed by `node.ExecCommandOnNode`?

Reply:
sorry , I meant the OR operator

Reply:
Jiri have noticed that using || was flaky - depending also on stalld running or not.
but since I have discovered an issue with the parrallel runs of go procedure not properly blocked by the wg.wait (fixed in the last commit here) this might have been the real issue. 
I will test the previous use of || true first and will re-introduce it if successful.

Reply:
@Tal-or  ok , I have tested this and it does fail . node.ExecCommandOnNode does not play nicely with the OR condition

Reply:
> This is not an actual piping BTW, just an OR operator. @jmencak Could you please elaborate when did you encounter issues with this `||` operator when it's part of a command executed by `node.ExecCommandOnNode`?

I don't exactly remember, but if you test (as Yanir did), you should see it.  The test was completely stuck depending on whether `stalld` was running.

Reply:
I'd strongly suggest reviewing the e2e tests for this as they might be the reason for the tests sometimes taking too long/timing out.

Reply:
What error do we get?
Then we should find another way catching error returning from `node.ExecCommandOnNode` 
Maybe we can postpone it to another PR

Reply:
The rest of the tests which are taking a significant chunck of time are tests pending on node reboots and waiting for mcp updates. we have a plan to reduce the wait time for mcps but this will be done in another PR after some desgin.
for the short term this PR should be merged + splitting the lanes further as suggested here : https://github.com/openshift/cluster-node-tuning-operator/pull/631

Reply:
> What error do we get?

In my case none.  It was just stuck/hanged forever until I pressed Ctrl+C.

Reply:
The error coming out from `node.ExecCommandOnNode(pidof stalld || true )` is: `timed out waiting for the condition`